docs/NewFlux_Comentary.md

```diff
@@ -0,0 +1,252 @@
+# New Flux
+
+Author: Nathaniel Cook
+
+Recently Paul Dix and I added a document on the vision of a new Flux that is more usable.
+The design there solves several key points of feedback we have received on Flux.
+
+
+That document describe the what and some of the why. This document will discuss the how.
+This document is not meant to be prescriptive rather just my personal commentary on how I see the vision of new Flux could be implemented.
+It is meant to be a seed to start a good team discussion around how  we can implement the new Flux vision.
+
+
+## What is new?
+
+First before we talk about how we would implement the new vision, we should clarify what is new.
+
+### Single Table vs Stream of Tables
+
+The new Flux data model changes to be a single table with a homogeneous schema.
+This is different from the current design which represents data as a stream (unbounded sequence) of tables with possibility differing schemas.
+
+This change is fundamental to the new design, the most common pit fall with current Flux is working with this stream of tables that may or may not be the same structure.
```

Comment:
This is a current pitfall of Flux? I'm not sure I understand the point made here.

Reply:
The ways this pitfall manifests is users write a Flux script they expect to work for data following a  certain shape. But the data shape isn't clear from just the script text so they often find the data has a different shape than they expected and so their script doesn't work. To make it even harder the shape of the data can change for each table, so while the script works for some of their data it doesn't work for all of it.

So given that a user needs to understand the shape of their data in order to write a good Flux script, the current design can change the shape of the data in unexpected ways. For example this query changes the group key three different times. Can you spot them?

```
from(bucket:"telegraf") 
    |> range(start: -1h)
    |> filter(fn: (r) => r.env == "acc")
    |> group(columns:["region", "env", "_measurement", "_field"])
    |> mean()
    |> map(fn: (r) => ({region: r.region, env: r.env, _measurement: r._measurement, _value: r._value * 2.0}))
    |> keep(columns:["region", "_value"])
    |> sum()
```

The calls to `group`, `map` and `keep` all change the shape of the data because they change the group key.

The end result is the user has to remember what there group key is and they have to follow the flow of the group key through their script in order to know what it is.